### ISSUE TITLE
Corrupt database, removeVertex gives NPE

### ISSUE THREAD

**Issue by RoarN** [orientechnologies/orientdb#6807]
using 2.0.15 on Windows
tested with 2.2.11 and had the same issue

We where deleting some vertexes in our database from our application when we got a NPE. After some testing we established that the NPE is from the OrientDB server; not our code.

```
2016-10-13 10:01:00,817 [qtp1656404914-1031] ERROR com.company.datastore.impl.blueprints.orientdb.administration.AdministrationProvider:1356 - Data Store exception in deleteUserCategory
java.lang.NullPointerException
	at com.tinkerpop.blueprints.impls.orient.OrientVertex.removeEdges(OrientVertex.java:322)
	at com.tinkerpop.blueprints.impls.orient.OrientVertex.removeInverseEdge(OrientVertex.java:446)
	at com.tinkerpop.blueprints.impls.orient.OrientVertex.removeEdges(OrientVertex.java:255)
	at com.tinkerpop.blueprints.impls.orient.OrientVertex.remove(OrientVertex.java:659)
	at com.tinkerpop.blueprints.impls.orient.OrientBaseGraph.removeVertex(OrientBaseGraph.java:758)
	at com.company.datastore.impl.blueprints.orientdb.administration.AdministrationProvider.__M_deleteUserCategory(AdministrationProvider.java:1349)
	at com.company.datastore.impl.blueprints.orientdb.administration.AdministrationProvider.deleteUserCategory(AdministrationProvider.java)
```

I opened the database in Studio to check how much had actually been deleted.
g.v('#50:11').out()

Only 2 records was still hanging on #50:11 and only one of them should be deleted, so I ran
delete vertex #53:63

and got an NPE error in Studio.

The OrientDB log reported:
2016-10-13 10:09:33:168 INFO  OrientDB Server v2.0.15 is active. [OServer]
2016-10-13 10:10:25:726 SEVERE Internal server error:
java.lang.NullPointerException [ONetworkProtocolHttpDb

The following steps has been tried to fix the problem.
1. export the database and import it into a fresh database -> same problem
2. tested in studio in latest version (2.2.11) -> same problem

**Comment by RoarN** [comment]
The corrupt database:

[datastore.zip](https://github.com/orientechnologies/orientdb/files/526623/datastore.zip)

**Comment by voondo** [comment]
Same problem here.

**Comment by lvca** [comment]
Checking it

**Comment by lvca** [comment]
Fixed in v2.2.x.

**Comment by RoarN** [comment]
That is great, now if I could actually use that version it would be fine. See #6404
